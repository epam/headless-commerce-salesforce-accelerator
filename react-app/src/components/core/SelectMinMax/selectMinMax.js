/*
 * Copyright 2020 EPAM Systems, Inc. (https://www.epam.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from "react";
import PropTypes from "prop-types";
import StyledTypography from "components/core/Typography";
import {
  StyledFormControl,
  StyledSelect,
  StyledSelectWrapper,
  StyledErrorMsg,
} from "./selectMinMaxStyled";
import ArrowIcon from "../ArrowIcon/index";

const SelectMinMax = ({
  IconComponent = ArrowIcon,
  handleChange = () => {},
  value,
  min,
  max,
  width,
  height,
  label,
  iconisrotated,
  error = false,
  errorMsg,
  fullWidth,
  margin,
  name = "",
  required,
}) => {
  const options = [];

  for (let i = min; i <= max; i += 1) {
    options.push(
      <option key={i} value={i}>
        {i}
      </option>
    );
  }

  return (
    <StyledSelectWrapper margin={margin}>
      <StyledTypography margin="0 0 4px 0">{label}</StyledTypography>
      <StyledFormControl
        variant="outlined"
        margin="dense"
        width={width}
        height={height}
        focused={false}
        error={error}
        fullWidth={fullWidth}
        required={required}
      >
        <StyledSelect
          native
          IconComponent={IconComponent}
          min={min.toString()}
          max={max.toString()}
          iconisrotated={iconisrotated?.toString()}
          onChange={handleChange}
          name={name}
          value={value}
        >
          {options}
        </StyledSelect>
      </StyledFormControl>
      {error && errorMsg ? <StyledErrorMsg>{errorMsg}</StyledErrorMsg> : null}
    </StyledSelectWrapper>
  );
};

SelectMinMax.propTypes = {
  IconComponent: PropTypes.elementType,
  handleChange: PropTypes.func,
  value: PropTypes.number,
  min: PropTypes.number,
  max: PropTypes.number,
  width: PropTypes.number,
  height: PropTypes.number,
  label: PropTypes.string,
  iconisrotated: PropTypes.bool,
  fullWidth: PropTypes.bool,
  margin: PropTypes.string,
  error: PropTypes.bool,
  errorMsg: PropTypes.string,
  name: PropTypes.string,
  required: PropTypes.bool,
};

export default SelectMinMax;
