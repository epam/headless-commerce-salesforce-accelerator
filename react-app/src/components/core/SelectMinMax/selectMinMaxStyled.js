/*
 * Copyright 2020 EPAM Systems, Inc. (https://www.epam.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import styled from "styled-components";
import { FormControl, Select } from "@material-ui/core";

export const StyledFormControl = styled(FormControl)`
  width: ${({ width = 120, fullWidth, theme }) =>
    fullWidth ? "100%" : theme.pxToRem(width)};
  height: ${({ height = 40, theme }) => theme.pxToRem(height)};
  margin: 0;

  .MuiOutlinedInput-root.Mui-error .MuiOutlinedInput-notchedOutline {
    border-color: ${({ theme }) => theme.colors.punch};
  }
`;

export const StyledSelect = styled(Select)`
  .MuiInputBase-input {
    font: ${({ theme }) => theme.font("regular", 14)};
    color: ${({ theme }) => theme.colors.mineShaft};
    background-color: ${({ theme }) => theme.colors.white};
  }

  .MuiSelect-icon {
    width: ${({ theme }) => theme.pxToRem(24)};
    height: ${({ theme }) => theme.pxToRem(24)};
    transform: ${({ iconisrotated }) =>
      iconisrotated === "true" ? "rotate(90deg)" : "none"};
  }

  option:disabled {
    color: ${({ theme }) => theme.colors.dimGray};
  }
`;

export const StyledSelectWrapper = styled.div`
  margin: ${({ margin }) => margin || "16px 0 0"};
`;

export const StyledErrorMsg = styled.div`
  font: ${({ theme }) => theme.font("regular", 12, 16)};
  color: ${({ theme }) => theme.colors.punch};
`;
