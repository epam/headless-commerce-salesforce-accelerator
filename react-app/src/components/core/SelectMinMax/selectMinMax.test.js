/*
 * Copyright 2020 EPAM Systems, Inc. (https://www.epam.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from "react";
import { shallow } from "enzyme";
import SelectMinMax from ".";
import { StyledFormControl, StyledSelect } from "./selectMinMaxStyled";

import getMainTheme from "../../../theme";

const defaultProps = {
  iconisrotated: true,
  min: 1,
  max: 10,
};

const theme = getMainTheme();

describe("SelectMinMax", () => {
  const wrapped = shallow(<SelectMinMax {...defaultProps} />);
  it("should match snapshot", () => {
    expect(wrapped).toMatchSnapshot();
  });
});

describe("SelectMinMax", () => {
  const wrapped = shallow(<SelectMinMax theme={theme} {...defaultProps} />);
  it("should match snapshot", () => {
    expect(wrapped.dive()).toMatchSnapshot();
  });
});

describe("StyledFormControl", () => {
  const wrapped = shallow(<StyledFormControl theme={theme} />);
  it("should match snapshot", () => {
    expect(wrapped.dive()).toMatchSnapshot();
  });
});

describe("StyledSelect", () => {
  const wrapped = shallow(<StyledSelect theme={theme} />);
  it("should match snapshot", () => {
    expect(wrapped.dive()).toMatchSnapshot();
  });
});
