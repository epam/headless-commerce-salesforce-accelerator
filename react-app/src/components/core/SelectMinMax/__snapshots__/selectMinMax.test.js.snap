// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectMinMax should match snapshot 1`] = `
<styled.div>
  <Styled(WithStyles(ForwardRef(Typography)))
    margin="0 0 4px 0"
  />
  <Styled(WithStyles(ForwardRef(FormControl)))
    error={false}
    focused={false}
    margin="dense"
    variant="outlined"
  >
    <Styled(WithStyles(ForwardRef(Select)))
      IconComponent={[Function]}
      iconisrotated="true"
      max="10"
      min="1"
      name=""
      native={true}
      onChange={[Function]}
    >
      <option
        key="1"
        value={1}
      >
        1
      </option>
      <option
        key="2"
        value={2}
      >
        2
      </option>
      <option
        key="3"
        value={3}
      >
        3
      </option>
      <option
        key="4"
        value={4}
      >
        4
      </option>
      <option
        key="5"
        value={5}
      >
        5
      </option>
      <option
        key="6"
        value={6}
      >
        6
      </option>
      <option
        key="7"
        value={7}
      >
        7
      </option>
      <option
        key="8"
        value={8}
      >
        8
      </option>
      <option
        key="9"
        value={9}
      >
        9
      </option>
      <option
        key="10"
        value={10}
      >
        10
      </option>
    </Styled(WithStyles(ForwardRef(Select)))>
  </Styled(WithStyles(ForwardRef(FormControl)))>
</styled.div>
`;

exports[`SelectMinMax should match snapshot 2`] = `
<div
  className="sc-AxhCb gJDamB"
>
  <Styled(WithStyles(ForwardRef(Typography)))
    margin="0 0 4px 0"
  />
  <Styled(WithStyles(ForwardRef(FormControl)))
    error={false}
    focused={false}
    margin="dense"
    variant="outlined"
  >
    <Styled(WithStyles(ForwardRef(Select)))
      IconComponent={[Function]}
      iconisrotated="true"
      max="10"
      min="1"
      name=""
      native={true}
      onChange={[Function]}
    >
      <option
        key="1"
        value={1}
      >
        1
      </option>
      <option
        key="2"
        value={2}
      >
        2
      </option>
      <option
        key="3"
        value={3}
      >
        3
      </option>
      <option
        key="4"
        value={4}
      >
        4
      </option>
      <option
        key="5"
        value={5}
      >
        5
      </option>
      <option
        key="6"
        value={6}
      >
        6
      </option>
      <option
        key="7"
        value={7}
      >
        7
      </option>
      <option
        key="8"
        value={8}
      >
        8
      </option>
      <option
        key="9"
        value={9}
      >
        9
      </option>
      <option
        key="10"
        value={10}
      >
        10
      </option>
    </Styled(WithStyles(ForwardRef(Select)))>
  </Styled(WithStyles(ForwardRef(FormControl)))>
</div>
`;

exports[`StyledFormControl should match snapshot 1`] = `
<ForwardRef(FormControl)
  className="sc-AxirZ dWHIYc"
  classes={
    Object {
      "fullWidth": "MuiFormControl-fullWidth",
      "marginDense": "MuiFormControl-marginDense",
      "marginNormal": "MuiFormControl-marginNormal",
      "root": "MuiFormControl-root",
    }
  }
  theme={
    Object {
      "breadcrumbsH": 40,
      "breakpoints": Object {
        "mobile": 992,
      },
      "colLg": 74,
      "colMd": 69,
      "colSm": 60,
      "colors": Object {
        "alabaster": "#FBFBFB",
        "amber": "#FFC000",
        "athensGray": "#FAFBFC",
        "black": "#000",
        "blue": "#376A9A",
        "dimGray": "#CBCDCE",
        "gullGrey": "#A7A9AA",
        "iron": "#D9DCDD",
        "lightGray": "#DEE2E3",
        "mineShaft": "#222222",
        "miscellaneous": "linear-gradient(90deg, #E77878 0%, #EFD063 25.00%, #65D5EE 50.00%, #C083E6 75.00%, #7F6EE8 100%);",
        "pale": "#FFEFEF",
        "punch": "#DA3232",
        "regentGrey": "#7D8081",
        "rollingStone": "#7E8182",
        "semiWhite": "rgba(255, 255, 255, 0.75)",
        "silverChalice": "#A9A9A9",
        "turmeric": "#C5D153",
        "wattle": "#CEDB56",
        "whisper": "#EBEBEB",
        "white": "#FFFFFF",
        "yellowGreen": "#D8E278",
      },
      "contentWidth": 1200,
      "font": [Function],
      "fontSize": 14,
      "footerH": 331,
      "gutter": 24,
      "headerH": 160,
      "navBarH": 56,
      "pxToRem": [Function],
      "searchBarH": 72,
      "topBarH": 32,
    }
  }
/>
`;

exports[`StyledSelect should match snapshot 1`] = `
<ForwardRef(Select)
  className="sc-AxiKw gNSnbf"
  classes={
    Object {
      "disabled": "Mui-disabled",
      "filled": "MuiSelect-filled",
      "icon": "MuiSelect-icon",
      "iconFilled": "MuiSelect-iconFilled",
      "iconOpen": "MuiSelect-iconOpen",
      "iconOutlined": "MuiSelect-iconOutlined",
      "nativeInput": "MuiSelect-nativeInput",
      "outlined": "MuiSelect-outlined",
      "root": "MuiSelect-root",
      "select": "MuiSelect-select",
      "selectMenu": "MuiSelect-selectMenu",
    }
  }
  theme={
    Object {
      "breadcrumbsH": 40,
      "breakpoints": Object {
        "mobile": 992,
      },
      "colLg": 74,
      "colMd": 69,
      "colSm": 60,
      "colors": Object {
        "alabaster": "#FBFBFB",
        "amber": "#FFC000",
        "athensGray": "#FAFBFC",
        "black": "#000",
        "blue": "#376A9A",
        "dimGray": "#CBCDCE",
        "gullGrey": "#A7A9AA",
        "iron": "#D9DCDD",
        "lightGray": "#DEE2E3",
        "mineShaft": "#222222",
        "miscellaneous": "linear-gradient(90deg, #E77878 0%, #EFD063 25.00%, #65D5EE 50.00%, #C083E6 75.00%, #7F6EE8 100%);",
        "pale": "#FFEFEF",
        "punch": "#DA3232",
        "regentGrey": "#7D8081",
        "rollingStone": "#7E8182",
        "semiWhite": "rgba(255, 255, 255, 0.75)",
        "silverChalice": "#A9A9A9",
        "turmeric": "#C5D153",
        "wattle": "#CEDB56",
        "whisper": "#EBEBEB",
        "white": "#FFFFFF",
        "yellowGreen": "#D8E278",
      },
      "contentWidth": 1200,
      "font": [Function],
      "fontSize": 14,
      "footerH": 331,
      "gutter": 24,
      "headerH": 160,
      "navBarH": 56,
      "pxToRem": [Function],
      "searchBarH": 72,
      "topBarH": 32,
    }
  }
/>
`;
